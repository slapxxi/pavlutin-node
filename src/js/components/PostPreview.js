import React from 'react';
import { Link } from 'react-router-dom';
import Markdown from 'react-markdown';
import Icon from 'react-fontawesome';
import Tags from './Tags';
import { toHumanReadableDate } from '../utils';

function PostPreview({ post }) {
  const URL = `/blog/${post.slug}`;
  return (
    <div className="post post_preview">
      <h2 className="post__title">
        <Link
          to={URL}
          title={`Read ${post.title}`}
        >
          {post.title}
        </Link>
      </h2>
      <div className="post__meta">
        <Icon name="clock-o" /> {' '}
        {toHumanReadableDate(post.createdAt)}
      </div>
      <Tags tags={post.tags} className="post__tags" />
      <div className="post__description">
        <Markdown source={post.description || "There's no description."} />
      </div>
      <div>
        <Link className="button" to={URL}>
          Keep Reading
        </Link>
      </div>
    </div>
  );
}

export default PostPreview;
