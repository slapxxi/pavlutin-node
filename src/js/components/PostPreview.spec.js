import React from 'react';
import render from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import PostPreview from './PostPreview';

const post = {
  id: 0,
  title: 'Test Post',
  content: 'empty',
  description: 'empty',
  author: 'bot',
  slug: 'test-post',
  tags: ['test', 'react'],
};

it('renders', () => {
  const tree = render.create(
    <Router>
      <PostPreview post={post} />
    </Router>,
  );
  expect(tree.toJSON()).toMatchSnapshot();
});

it('renders with active tag', () => {
  const tree = render.create(
    <Router>
      <PostPreview post={post} activeTag="tag" />
    </Router>,
  );
  expect(tree.toJSON()).toMatchSnapshot();
});
