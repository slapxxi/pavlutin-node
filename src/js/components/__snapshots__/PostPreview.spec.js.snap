// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="preview"
>
  <header
    className="preview__header"
  >
    <h2
      className="preview__title"
    >
      <a
        href="/blog/test-post"
        onClick={[Function]}
        title="Read Test Post"
      >
        Test Post
      </a>
    </h2>
    <div
      className="preview__meta"
    >
      <span
        aria-hidden={true}
        className="fa fa-clock-o"
      />
       
       
      a few seconds ago
    </div>
    <ul
      className="tags preview__tags"
    >
      <li
        className="tags__tag"
      >
        <a
          href="/blog/tag/test"
          onClick={[Function]}
        >
          test
        </a>
      </li>
      <li
        className="tags__tag"
      >
        <a
          href="/blog/tag/react"
          onClick={[Function]}
        >
          react
        </a>
      </li>
    </ul>
  </header>
  <div
    className="preview__description"
  >
    <div>
      <p>
        empty
      </p>
    </div>
  </div>
  <div
    className="preview__button"
  >
    <a
      href="/blog/test-post"
      onClick={[Function]}
    >
      Keep Reading
    </a>
  </div>
</div>
`;

exports[`renders with active tag 1`] = `
<div
  className="preview"
>
  <header
    className="preview__header"
  >
    <h2
      className="preview__title"
    >
      <a
        href="/blog/test-post"
        onClick={[Function]}
        title="Read Test Post"
      >
        Test Post
      </a>
    </h2>
    <div
      className="preview__meta"
    >
      <span
        aria-hidden={true}
        className="fa fa-clock-o"
      />
       
       
      a few seconds ago
    </div>
    <ul
      className="tags preview__tags"
    >
      <li
        className="tags__tag"
      >
        <a
          href="/blog/tag/test"
          onClick={[Function]}
        >
          test
        </a>
      </li>
      <li
        className="tags__tag"
      >
        <a
          href="/blog/tag/react"
          onClick={[Function]}
        >
          react
        </a>
      </li>
    </ul>
  </header>
  <div
    className="preview__description"
  >
    <div>
      <p>
        empty
      </p>
    </div>
  </div>
  <div
    className="preview__button"
  >
    <a
      href="/blog/test-post"
      onClick={[Function]}
    >
      Keep Reading
    </a>
  </div>
</div>
`;
